src/lib/OpenEXR/ImfStandardAttributes.h

```diff
@@ -299,17 +762,103 @@ IMF_STD_ATTRIBUTE_DEF (framesPerSecond, FramesPerSecond, Rational)
 
 
 //
-// multiView -- defines the view names for multi-view image files.
+// multiView -- defines the view names for multi-view image files
+//
 // A multi-view image contains two or more views of the same scene,
 // as seen from different viewpoints, for example a left-eye and
 // a right-eye view for stereo displays.  The multiView attribute
 // lists the names of the views in an image, and a naming convention
 // identifies the channels that belong to each view.
 //
+// Files conformant to SMPTE ST 2065-4 ("ACES Container File Format")
+// which have a multiView attribute will always have the two strings
+// "right" and "left" in that order. When an ACES Container file is
+// used to hold a stereoscopic image, this attribute will be present.
+//
 // For details, please see header file ImfMultiView.h
 //
 
-IMF_STD_ATTRIBUTE_DEF (multiView , MultiView, StringVector)
+IMF_STD_ATTRIBUTE_DEF (multiView, MultiView, StringVector)
+
+
+//
+// framelines -- set of rectangles used for creative purposes, e.g. to
+// indicate safe areas, cropping for derivative image sequences, etc
+//
+// The way in which the format encodes these rectangles is not defined;
+// the encoding might be camera-vendor-specific, or it might be formally
+// or informally standardized (e.g. an ASC Frameline Decision List)
+//
+
+IMF_STD_ATTRIBUTE_DEF (framelines, Framelines, std::string)
+
+
+//
+// imageRotation -- rotation of the image in degrees, in the closed interval
+// [-45, +45], with the rotation around the center of the display window, a
+// positive value specifying a rotation from x-axis to y-axis
+//
+
+IMF_STD_ATTRIBUTE_DEF (imageRotation, ImageRotation, float)
+
+
+//
+// cameraPosition -- x, y, z position of the camera, in meters
+//
+// The attribute may record the center of the camera's sensor, or the
+// center of the entrance pupil of the lens. The coordinate system for
+// the camera's position is fixed in relation to the set, and is a
+// right-handed Cartesian coordinate system with the z-axis pointing
+// upwards and the y-axis pointing 90 degrees to the left of the
+// x-axis direction.
```

Comment:
This contradicts our usage in OpenEXR via Math.

```
//  The frustum is always located with the eye point at the
//  origin facing down -Z. This makes the Frustum class
//  compatable with OpenGL (or anything that assumes a camera
//  looks down -Z, hence with a right-handed coordinate system)
//  but not with RenderMan which assumes the camera looks down
//  +Z. Additional functions are provided for conversion from
//  and from various camera coordinate spaces.
```

Is this deviation from OpenEXR part of the 2065-4 spec? If so, we should add additional notes, including perhaps a matrix that converts to and from the ACES camera space.

Reply:
I will have a go at this in a bit but note this from the spec:

![image](https://user-images.githubusercontent.com/1134845/103467111-a99b9900-4d19-11eb-8743-da8c9a003658.png)

That 'or' in the first sentence is a deal-breaker for me, and I am going to have to bring this up in the SMPTE drafting group. If you can't even tell whether your camera position represents the sensor (presumably the point where the optical axis would intersect the front of the sensor) vs. the entrance pupil, then entrancePupilOffset becomes meaningless. Or, alternatively, one drops _"the center of the camera's sensor, or"_ and there's no need any longer for `entrancePupilOffset`...but that breaks compatibility, seriously, with anyone who was using the first meaning from ST 2065-4:2013; moreover, I suspect people are much more capable of taking a tape and reading out a distance from the mark on an outside of a camera body that represents the focal plane than they are in locating the entrance pupil on-the-fly, given that just the act of focusing can move it.

I take it when you say "via Math" you mean "via iMath"? And are you talking about this part of ImathFrustrum.h?

![image](https://user-images.githubusercontent.com/1134845/103467260-47439800-4d1b-11eb-8ab7-768f302aae47.png)

Reply:
Imath seems to support either row-matrix or matrix-column order. If one were providing examples of matrixes to go from ACES-style coordinates for camera position to OpenEXR style coordinates, which style is most consistent with the rest of OpenEXR?

Reply:
Imath's matrices have the memory layout where, if the floats were a 1D array, the translation part would be [12], [13], and [14].

Reply:
So, either, the matrix is stored in row major order but the vectors are column vectors, OR the matrix is stored in column major order but the vectors are row vectors. Those are equivalent. (But that is NOT the same as RMO+RV or CMO+CV, both of which are transposed compared to Imath/OpenEXR.)

Reply:
If it helps, OpenEXR/Imath's memory layout is the same as OpenGL and RenderMan.

Reply:
I think we should take a strong position of measuring distances from the sensor center, as the entrance pupil can only be determined by the sort of nerd with the sort of equipment that are not often going to be on set simultaneously. It's valuable to have the entrance pupil, but not as the fraying rope all the other measurements dangle from.

Reply:
This needs to be translated into comments if correct; it's from a document I have that summarizes changes to be made to 2065-4. Does it look/sound right?

![image](https://user-images.githubusercontent.com/1134845/103845261-b695f080-5069-11eb-9bb2-72ba196a98b7.png)

Reply:
Incidentally, and this shows how long I've been out of production, in the real world, what fraction of OpenEXR frames have either or both of `worldToCamera` or `worldToNDC` defined?

Reply:
Your proposed revisions are lovely, thank you. I don't know the answer to the real world question, as I don't have any OpenEXR files that record either. My role in this conversation is semi-pedantic bystander and pest ;)